python/ray/worker.py

```diff
@@ -424,12 +424,14 @@ def get_gpu_ids():
 
     # TODO(ilr) Handle inserting resources in local mode
     all_resource_ids = global_worker.core_worker.resource_ids()
-    assigned_ids = []
+    assigned_ids = set()
```

Comment:
@richardliaw, kind of an edge case, but the order of the devices in CUDA_VISIBLE_DEVICES actually does have some implications. Namely, since CUDA assigns IDs based on the fastest device first (see [CUDA_DEVICE_ORDER](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars)), you usually want to use device 0 first when available.

One alternative might be to use an order-preserving dedup instead: https://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-whilst-preserving-order/39835527#39835527

Reply:
hmm, what if we just sort the returned list at 434? I think Ray doesn't have any ordering guarantees from the detected CUDA devices, though maybe we can support that later

Reply:
Yeah, that works!  May want to double check if the ids here are strings or ints for that, though (would want to sort lexicographically for strings).